hw/ip/spi_host/data/spi_host_testplan.hjson

```diff
@@ -66,68 +61,140 @@
             '''
       milestone: V2
-      tests: ["spi_host_tx_overflow_error"]
+      tests: []
     }
     {
       name: component_reset
       desc: '''
             Test components (spi_fsm, rx_fifo, tx_fifo) are randomly reset
 
             Stimulus:
-              - Reset the components of spi_host randomly after a random number of data shows up on fifos
+              - Reset the components of spi_host randomly
+                after a random number of data shows up on fifos
 
             Checking:
               - Ensure that reads to RXDATA register yield 0s after the rx_fifo is reset
               - Ensure that transactions are dropped in both the scoreboard and spi_agent monitor
                 after the tx_fifo or spi_fsm is reset
             '''
       milestone: V2
-      tests: ["spi_host_component_reset"]
+      tests: []
     }
     {
-      name: clock_domain
+      name: modes
       desc: '''
-            TBD - Verify the function of clock-crossing domain for spi_host
+            - Verify the function of spi_host in different special_modes
+              (pass-through and manual_ctrl_cs)
 
             Stimulus:
               - TBD
-
             Checking:
               - TBD
       '''
       milestone: V2
-      tests: ["spi_host_clock_domain"]
+      tests: []
     }
     {
-      name: special_modes
+      name: cpol_cpha
       desc: '''
-            TBD - Verify the function of spi_host in special_modes (passthrough and manual_ctrl_cs)
+            Stimulus:
+              - Randomly chip select for different polarity / phase
 
+            Checking:
+              - Check that the DUT operates correctly under different cs_n settings
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: full_cycle
+      desc: '''
             Stimulus:
-              - TBD
+              - randomly select FULLCYC to be set
 
             Checking:
-              - TBD
-      '''
+              - Check that the data can be read one full cycle after the data was asserted
+            '''
       milestone: V2
-      tests: ["spi_host_special_modes"]
+      tests: []
+    }
+    {
+      name: endian
+      desc: '''
+            Stimulus:
+              - set the compile time parameter to configure the DUT in both modes.
+              if possible this should be randomized for all test to verify support in both modes
+
+            Checking:
+              - Check all features are supported correctly in both little and big endian mode
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: duplex
+      desc: '''
+            Stimulus:
+              - in standard mode set the DUT to run full duplex
+
+            Checking:
+              - verify that the DUT support both half and full duplex in standard mode.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tx_only
```

Comment:
I think we need something for rx-only segments as well (to ensure that tx data isn't consumed during those)

Reply:
@martin-lueker at first I was thinking that it would not be needed.
but on second thought I think it is a good idea. and certainly can hurt. I have updated the test point to include both